app/controllers/merchants_controller.rb

```diff
@@ -1,11 +1,11 @@
 class MerchantsController < ApplicationController
 
   def index
-    @merchants = User.all_active_merchants
+    @merchants = User.all_active_merchants if !current_admin?
+    @merchants = User.where(role: :merchant) if current_admin?
```

Comment:
I think we will try to move this out of the business logic if we have enough time but it's certainly ok for right now.